recipes/ogre/1.x/conanfile.py

```diff
@@ -0,0 +1,290 @@
+import conan.tools.files
+import textwrap, shutil
+
+class OgreCmakeConan(ConanFile):
+    name = "ogre"
+    license = "MIT"
+    homepage = "https://github.com/OGRECave/ogre"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "A scene-oriented, flexible 3D engine written in C++ "
+    topics = ("graphics", "rendering", "engine", "c++")
+
+    settings = "os", "compiler", "build_type", "arch"
+
+    generators = "cmake", "cmake_find_package"
+    exports_sources = "CMakeLists.txt", "patches/**"
+    #options copied from https://github.com/StatelessStudio/ogre-conan/blob/master/conanfile.py
+    options = {
+        "shared": [True, False],
+        "fPIC": [True, False],
+        "cp_bin_dir": "ANY",
+        "cp_media_dir": "ANY",
+        "disable_plugins": [True, False],
+        "OGRE_ASSERT_MODE": "ANY",
+        "OGRE_BUILD_COMPONENT_BITES": [True, False],
+        "OGRE_BUILD_COMPONENT_CSHARP": [True, False],
+        "OGRE_BUILD_COMPONENT_HLMS": [True, False],
+        "OGRE_BUILD_COMPONENT_JAVA": [True, False],
+        "OGRE_BUILD_COMPONENT_MESHLODGENERATOR": [True, False],
+        "OGRE_BUILD_COMPONENT_OVERLAY": [True, False],
+        "OGRE_BUILD_COMPONENT_OVERLAY_IMGUI": [True, False],
+        "OGRE_BUILD_COMPONENT_PAGING": [True, False],
+        "OGRE_BUILD_COMPONENT_PROPERTY": [True, False],
+        "OGRE_BUILD_COMPONENT_PYTHON": [True, False],
+        "OGRE_BUILD_COMPONENT_RTSHADERSYSTEM": [True, False],
+        "OGRE_BUILD_COMPONENT_TERRAIN": [True, False],
+        "OGRE_BUILD_COMPONENT_VOLUME": [True, False],
+        "OGRE_BUILD_DEPENDENCIES": [True, False],
+        "OGRE_BUILD_PLUGIN_BSP": [True, False],
+        "OGRE_BUILD_PLUGIN_DOT_SCENE": [True, False],
+        "OGRE_BUILD_PLUGIN_OCTREE": [True, False],
+        "OGRE_BUILD_PLUGIN_PCZ": [True, False],
+        "OGRE_BUILD_PLUGIN_PFX": [True, False],
+        "OGRE_BUILD_PLUGIN_STBI": [True, False],
+        "OGRE_BUILD_RENDERSYSTEM_D3D11": [True, False],
+        "OGRE_BUILD_RENDERSYSTEM_GL": [True, False],
+        "OGRE_BUILD_RENDERSYSTEM_GL3PLUS": [True, False],
+        "OGRE_BUILD_RTSHADERSYSTEM_SHADERS": [True, False],
+        "OGRE_BUILD_SAMPLES": [True, False],
+        "OGRE_BUILD_TESTS": [True, False],
+        "OGRE_BUILD_TOOLS": [True, False],
+        "OGRE_CONFIG_ENABLE_QUAD_BUFFER_STEREO": [True, False],
+        "OGRE_CONFIG_FILESYSTEM_UNICODE": [True, False],
+        "OGRE_CONFIG_THREADS": "ANY",
+        "OGRE_CONFIG_THREAD_PROVIDER": "ANY",
+        "OGRE_ENABLE_PRECOMPILED_HEADERS": [True, False],
+        "OGRE_INSTALL_PDB": [True, False],
+        "OGRE_INSTALL_SAMPLES": [True, False],
+        "OGRE_INSTALL_TOOLS": [True, False],
+        "OGRE_INSTALL_VSPROPS": [True, False],
+        "OGRE_NODELESS_POSITIONING": [True, False],
+        "OGRE_PROFILING_REMOTERY_PATH": "ANY",
+        "OGRE_RESOURCEMANAGER_STRICT": "ANY",
+        "ogre_static": [True, False],
```

Comment:
```suggestion
```

Reply:
@ggulgulia please, consider this suggestion. the option `shared`is used to switch between static/shared library.